File: src/main/tests/pythonGenerated/factorial.py
import typing

def factorial(n) -> int:
    if n <= 1:
        return 1
    return n * (factorial(n-1))

def main() -> int:
    num: int = 5
    res = factorial(num)
    print("Factorial of ", num, " is ", res, sep="")
    print()
    return 0

if __name__ == "__main__":
    main()

File: src/main/tests/input/factorial.cpp
#include <iostream>

int factorial(int n) {
    if (n <= 1){
        return 1;
    }
    return n * (factorial (n - 1));
}

int main() {
    int num = 5;
    auto res = factorial(num);
    std::cout << "Factorial of " << num << " is " << res << std::endl;
    return 0;
}